beets/util/__init__.py

```diff
@@ -636,7 +657,10 @@ def unique_path(path):
 ]
 
 
-def sanitize_path(path, replacements=None):
+def sanitize_path(
+        path: Bytestring,
+        replacements: Optional[Sequence[Sequence[Pattern | str]]] = None,
+) -> str | bytes:
```

Comment:
Shouldn't this be `str -> str` ?

Reply:
If that's the case then there are flow-in effects. The `path` variable goes through `_legalize_stage()` and `legalize_path()` without being changed, so if `sanitize_path` only accepts strings, then so must those functions to keep it consistent.

I've made these changes in 9453a513